pkg/webhook/workloadspread/validating/workloadspread_validation.go

```diff
@@ -0,0 +1,253 @@
+						allErrs = append(allErrs, field.Invalid(fldPath.Index(i).Child("requiredNodeSelectorTerm"), subset.RequiredNodeSelectorTerm, fmt.Sprintf("Convert_v1_NodeSelectorTerm_To_core_NodeSelectorTerm failed: %v", err)))
+					} else {
+						allErrs = append(allErrs, corevalidation.ValidateNodeSelectorTerm(*coreNodeSelectorTerm, fldPath.Index(i).Child("requiredNodeSelectorTerm"))...)
+					}
+				}
+
+				if subset.PreferredNodeSelectorTerms != nil {
+					corePreferredSchedulingTerms := make([]core.PreferredSchedulingTerm, 0, len(subset.PreferredNodeSelectorTerms))
+					for i, term := range subset.PreferredNodeSelectorTerms {
+						corePreferredSchedulingTerm := &core.PreferredSchedulingTerm{}
+						if err := corev1.Convert_v1_PreferredSchedulingTerm_To_core_PreferredSchedulingTerm(term.DeepCopy(), corePreferredSchedulingTerm, nil); err != nil {
+							allErrs = append(allErrs, field.Invalid(fldPath.Index(i).Child("preferredSchedulingTerms"), subset.PreferredNodeSelectorTerms, fmt.Sprintf("Convert_v1_PreferredSchedulingTerm_To_core_PreferredSchedulingTerm failed: %v", err)))
+						} else {
+							corePreferredSchedulingTerms = append(corePreferredSchedulingTerms, *corePreferredSchedulingTerm)
+						}
+					}
+
+					allErrs = append(allErrs, corevalidation.ValidatePreferredSchedulingTerms(corePreferredSchedulingTerms, fldPath.Index(i).Child("preferredSchedulingTerms"))...)
+				}
+			}
+
+			if subset.Tolerations != nil {
+				var coreTolerations []core.Toleration
+				for i, toleration := range subset.Tolerations {
+					coreToleration := &core.Toleration{}
+					if err := corev1.Convert_v1_Toleration_To_core_Toleration(&toleration, coreToleration, nil); err != nil {
+						allErrs = append(allErrs, field.Invalid(fldPath.Index(i).Child("tolerations"), subset.Tolerations, fmt.Sprintf("Convert_v1_Toleration_To_core_Toleration failed: %v", err)))
+					} else {
+						coreTolerations = append(coreTolerations, *coreToleration)
+					}
+				}
+				allErrs = append(allErrs, corevalidation.ValidateTolerations(coreTolerations, fldPath.Index(i).Child("tolerations"))...)
+			}
+
+			if subset.MaxReplicas != nil {
+				if i < len(subsets)-1 {
+					subsetMaxReplicas, err := intstr.GetValueFromIntOrPercent(subset.MaxReplicas, 100, true)
+					if err != nil || subsetMaxReplicas < 0 {
+						allErrs = append(allErrs, field.Invalid(fldPath.Index(i).Child("maxReplicas"), subset.MaxReplicas, "maxReplicas is not valid for subset"))
+					}
+					if subset.MaxReplicas.Type == intstr.String && subsetMaxReplicas > 100 {
+						allErrs = append(allErrs, field.Invalid(fldPath.Index(i).Child("maxReplicas"), subset.MaxReplicas, "maxReplicas exceeds 100% for subset"))
+					}
+				} else {
+					// validate maxReplicas of the end subset
+					allErrs = append(allErrs, field.Invalid(fldPath.Index(i).Child("maxReplicas"), subset.MaxReplicas, "maxReplicas of end subset must be nil"))
```

Comment:
is the last subset valid if maxreplicas is specifed as percent?

Reply:
We enforce user to set the last subset' maxReplicas with nil.